ledger/src/leader_schedule.rs

```diff
@@ -48,6 +64,28 @@ impl LeaderSchedule {
     pub fn num_slots(&self) -> usize {
         self.slot_leaders.len()
     }
+
+    /// Returns indices starting from 'offset' where pubkey is the leader.
+    pub(crate) fn get_schedule(
+        &self,
+        pubkey: &Pubkey,
+        offset: usize, // Starting index.
+    ) -> impl Iterator<Item = usize> {
+        let index = self.index.get(pubkey).cloned().unwrap_or_default();
+        let num_slots = self.slot_leaders.len();
+        let size = index.len();
+        #[allow(clippy::reversed_empty_ranges)]
+        let range = if index.is_empty() {
+            1..=0 // Intentionally empty range of type RangeInclusive.
+        } else {
+            let offset = index
+                .binary_search(&(offset % num_slots))
```

Comment:
`offset % num_slots` is for future proofing right? I think today it should always be true `offset < num_slots` since `num_slots` is always an epoch number of slots.